src/client/extension.ts

```diff
@@ -15,8 +16,9 @@ export function activate(context: vscode.ExtensionContext) {
 	let testFile = vscode.commands.registerCommand('v.test.file',commands.testFile);
 	let testPackage = vscode.commands.registerCommand('v.test.package',commands.testPackage);
 	let playground = vscode.commands.registerCommand('v.playground',commands.playground);
-	
+
 	registerFormatter();
+	vscode.window.onDidCloseTerminal(handle)
```

Comment:
You could also replace it with a function, e.g. `attachTerminalCloseListener`, and put the block:
```ts
// run.ts
export function attachTerminalCloseListener() {
	vscode.window.onDidCloseTerminal(term => {
		if (term.name == 'V') { 
			vRunTerm = null; 
		} 
	});
}
```
inside that new function.

Reply:
Great idea @alexesprit !!!